`@wordpress/api-fetch` works weirdly with ES modules

When using `@wordpress/api-fetch` in a TypeScript library, `tsc` complains that `apiFetch` is not callable.

The problem being encountered here is described in depth at https://github.com/arethetypeswrong/arethetypeswrong.github.io/blob/main/docs/problems/CJSOnlyExportsDefault.md

### Example

<details><summary>package.json</summary>

```json
{
	"type": "module",
	"dependencies": {
		"@wordpress/api-fetch": "^6.48.0",
		"typescript": "^5.0.4"
	}
}
```

</details>

<details><summary>tsconfig.json</summary>

```jsonc
{
	"include": [ "./src/**/*" ],
	"compilerOptions": {
		"outDir": "./build/",
		// Per https://www.typescriptlang.org/docs/handbook/modules/theory.html#module-resolution-for-libraries
		"module": "nodenext",
		"moduleResolution": "nodenext"
	}
}
```

</details>

<details><summary>src/index.ts</summary>

```ts
import apiFetch from '@wordpress/api-fetch';

apiFetch( { path: '/wp/v2/posts' } ).then( ( posts ) => {
	console.log( posts );
} );
```

</details>

Run `npm install`, then `npm exec tsc` to attempt to build the "library".

### Expected results

Build succeeds

### Actual results

```
src/index.ts:3:1 - error TS2349: This expression is not callable.
  Type 'typeof import("/tmp/test/node_modules/@wordpress/api-fetch/build-types/index")' has no call signatures.

3 apiFetch( { path: '/wp/v2/posts' } ).then( ( posts ) => {
  ~~~~~~~~


Found 1 error in src/index.ts:3
```

### Notes

If we change the `apiFetch()` call to `apiFetch.default()`, `tsc` is happy but then the code won't work when the library is bundled and served to a browser.

To make things work in both environments, we needed to do something like [this](https://github.com/Automattic/jetpack/blob/f9312c73ee6f00465d656718638116039a950ce4/projects/js-packages/ai-client/src/api-fetch/index.ts).

--- a/packages/api-fetch/src/index.ts
+++ b/packages/api-fetch/src/index.ts
@@ -157,7 +157,7 @@ export interface ApiFetch {
  * @param options The options for the fetch.
  * @return A promise representing the request processed via the registered middlewares.
  */
-export const apiFetch: ApiFetch = ( options ) => {
+const apiFetch: ApiFetch = ( options ) => {
 	// creates a nested function chain that calls all middlewares and finally the `fetchHandler`,
 	// converting `middlewares = [ m1, m2, m3 ]` into:
 	// ```

--- a/packages/api-fetch/src/test/exports.test.ts
+++ b/packages/api-fetch/src/test/exports.test.ts
@@ -1,11 +1,7 @@
-import apiFetch, { apiFetch as namedApiFetch } from '../index';
+import apiFetch from '..';
 
 describe( 'apiFetch exports', () => {
 	it( 'default export is callable', () => {
 		expect( typeof apiFetch ).toBe( 'function' );
-	} );
-
-	it( 'named export is callable', () => {
-		expect( typeof namedApiFetch ).toBe( 'function' );
 	} );
 } );

